utils/ctranslate.py

```diff
@@ -0,0 +1,62 @@
+import csv
+import json
+import os
+import time
+
+import ctranslate2
+import sentencepiece
+
+import utils.benchmark as bench_utils
+from utils.misc import advertise_aio
+
+class CTranslateRunner:
+    """
+    A class providing facilities to run CTranslate2 model.
+    """
+
+    def __init__(self, model, tokenizer):
+        # try:
+        #     #TODO: Check for AIO
+        # except AttributeError:
+        #     utils.advertise_aio("CTranslate2")
+
+        self.translator = ctranslate2.Translator(model, device='cpu', compute_type='auto', inter_threads=1, intra_threads=bench_utils.get_intra_op_parallelism_threads())
+        self.tokenizer = sentencepiece.SentencePieceProcessor(tokenizer)
+
+        self.__times_invoked = 0
+        self.__start_times = list()
+        self.__finish_times = list()
+
+        print("\nRunning with CTranslate2\n")
+
+    def run(self, input):
+        start = time.time()
+        outputs = self.translator.translate_batch(input)
+        finish = time.time()
+
+        self.__start_times.append(start)
+        self.__finish_times.append(finish)
+        self.__times_invoked += 1
+
+        return outputs
+
+    def print_performance_metrics(self, batch_size):
+        """
+        A function printing performance metrics on runs executed by the runner so far.
+        :param batch_size: int, batch size - if batch size was varying over the runs an average should be supplied
+        """
+        perf = bench_utils.print_performance_metrics(
+            self.__start_times, self.__finish_times, self.__times_invoked, batch_size)
+        if os.getenv("AIO_PROFILER", "0") == "1":
+            ctranslate2.AIO.print_profile_data() #TODO: See if it works
```

Comment:
Does it?

Reply:
This doesn't work, .AIO module is not available in ctranslate2-aio. But it's not needed as AIO profiler works nonetheless simply due to AIO_PROFILER sys env being set. So this can be removed.